Resources/Prototypes/Floof/Entities/Objects/Weapons/guns.yml

```diff
@@ -48,6 +48,26 @@
     # zeroVisible: true
   # - type: Appearance
 
+- type: entity
+  name: borg pulse rifle
+  parent: WeaponPulseRifle
+  id: BorgWeaponPulseRifle
+  description: A weapon that is almost as infamous as its users.
+  components:
+  - type: BatterySelfRecharger
+    autoRecharge: true
+    autoRechargeRate: 30  
+
+- type: entity
+  name: borg Xray Cannon
```

Comment:
keep consistency with naming conventions with regards to display names, lowercase and hyphen x-ray.